spec/system/tech_shop_spec.rb

```diff
@@ -37,150 +39,142 @@
       results = Product.search('IPHONE')
-      expect(results.size).to eq(1)
-      expect(results).to include(@iphone)
+      expect(results).to contain_exactly(@iphone)
     end
 
     it 'finds all amazon products' do
       results = Product.search('amazon')
-      expect(results.size).to eq(3)
-      expect(results).to include(@android, @samsung, @motorola)
+      expect(results).to contain_exactly(@android, @samsung, @motorola)
     end
 
     it 'finds all "palm" phones with wildcard word search' do
       results = Product.search('pal')
-      expect(results.size).to eq(2)
-      expect(results).to include(@palmpre, @palm_pixi_plus)
+      expect(results).to contain_exactly(@palmpre, @palm_pixi_plus)
     end
 
     it 'searches multiple words from the same field' do
       results = Product.search('palm pixi plus')
-      expect(results.size).to eq(1)
-      expect(results).to include(@palm_pixi_plus)
+      expect(results).to contain_exactly(@palm_pixi_plus)
     end
 
     it 'finds using phrase search' do
       results = Product.search('coco "palm"')
-      expect(results.size).to eq(1)
-      expect(results).to include(@palm_pixi_plus)
+      expect(results).to contain_exactly(@palm_pixi_plus)
     end
 
     it 'narrows the results by searching across multiple fields' do
       results = Product.search('apple iphone')
-      expect(results.size).to eq(2)
-      expect(results).to include(@iphone)
+      expect(results).to include(@iphone, @macbook)
     end
 
     it 'does not search on non-indexed fields' do
-      results = Product.search('features')
-      expect(results.size).to eq(0)
+      expect(Product.search('features')).to be_empty
     end
 
-    it 'deletes the associated record' do
+    it 'deletes associated document on #destroy' do
       ipad = Product.create!(name: 'ipad', href: 'apple', tags: ['awesome', 'great battery'],
                              description: 'Big screen')
 
       ipad.index!(true)
       results = Product.search('ipad')
-      expect(results.size).to eq(1)
+      expect(results).to contain_exactly(ipad)
 
       ipad.destroy
-      results = Product.search('ipad')
-      expect(results.size).to eq(0)
+      AsyncHelper.await_last_task
+
+      results = Product.raw_search('ipad')['hits']
+      expect(results).to be_empty
     end
 
     context 'when a document cannot be found in ActiveRecord' do
       it 'does not throw an exception' do
         Product.index.add_documents!(@palmpre.attributes.merge(id: -1))
-        expect { Product.search('pal').to_json }.not_to raise_error
+        expect { Product.search('pal') }.not_to raise_error
         Product.index.delete_document!(-1)
       end
 
-      it 'returns the other results if those are still available locally' do
+      it 'returns other available results' do
         Product.index.add_documents!(@palmpre.attributes.merge(id: -1))
-        expect(JSON.parse(Product.search('pal').to_json).size).to eq(2)
```

Comment:
I do not understand why this was being converted to `json` and back. If there's some edge case or special significance to this please let me know.